libs/SalesforceNetwork/SalesforceNetwork/SalesforceNetwork/Classes/Network/Action/CSFAction.m

```diff
@@ -535,21 +535,24 @@ - (void)completeOperationWithError:(NSError *)error {
 
 - (id)contentFromData:(NSData*)data fromResponse:(NSHTTPURLResponse*)response error:(NSError**)error {
     id content = nil;
-    
+    BOOL requestSucceeded = (response.statusCode >= 200 && response.statusCode < 300);
+
+    // try to parse response if response is not an error or response status code is between the specified status code range
+    // 2xx is for successful request
+    // 4xx is for client error that may contains valuable error information in response
     NSError *jsonParseError = nil;
     if ([self.responseData length] > 0) {
```

Comment:
Good question, I honestly don't remember if we did it for a specific purpose but I agree with your proposal. Let me dig more in case I find something but otherwise go ahead